### PR TITLE
Submission for io.github.daviddesimone.opencloudsaves 

### PR DESCRIPTION
**Pull request by DavidDeSimone**
I am attempting to add my application, https://github.com/DavidDeSimone/OpenCloudSaves. This is an application that syncs video game save data to various cloud providers. 

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)* (SEE NOTE 1)
- [x] All assets referenced in the manifest are redistributable by any party.  ~~If not, the unredistributable parts are using an extra-data source type.~~
- [x] I am an upstream contributor to the project. ~~If not, I contacted upstream developers about submitting their software to Flathub. **Link:**~~
- [x] ~~I own the domain used in the application ID or~~ the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

NOTE 1:  Users can enter paths to save data stored on their computers to be uploaded to a custom cloud. Due to this, I need `--filesystem=host'` since save files will live on the host file system as arbitrary locations based on the user's preferences for save data locations.

### CODE REVIEW

File: io.github.daviddesimone.opencloudsaves.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    build-commands:
+      - $GOROOT/bin/go build -trimpath -o opencloudsave
+      - install -D opencloudsave /app/bin/opencloudsave
+      - install -D ./build/linux/bootstrap-opencloudsave.sh /app/bin/bootstrap-opencloudsave.sh
+      - mkdir -p /app/share/applications/
+      - install -D ./build/linux/icon/io.github.daviddesimone.opencloudsaves.512.png /app/share/icons/hicolor/512x512/apps/io.github.daviddesimone.opencloudsaves.png
+      - install -D ./build/linux/icon/io.github.daviddesimone.opencloudsaves.256.png /app/share/icons/hicolor/256x256/apps/io.github.daviddesimone.opencloudsaves.png
+      - install -D ./build/linux/icon/io.github.daviddesimone.opencloudsaves.128.png /app/share/icons/hicolor/128x128/apps/io.github.daviddesimone.opencloudsaves.png
+      - install -D ./build/linux/icon/io.github.daviddesimone.opencloudsaves.64.png /app/share/icons/hicolor/64x64/apps/io.github.daviddesimone.opencloudsaves.png
+      - install -D ./build/linux/icon/io.github.daviddesimone.opencloudsaves.32.png /app/share/icons/hicolor/32x32/apps/io.github.daviddesimone.opencloudsaves.png
```
Comment (barthalion) [inline io.github.daviddesimone.opencloudsaves.yml:57]: Unless they're used internally by the app, just 512x512 would do the trick.

File: io.github.daviddesimone.opencloudsaves.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
+app-id: io.github.daviddesimone.opencloudsaves
+base: io.elementary.BaseApp
+base-version: juno-21.08
```
Comment (barthalion) [inline io.github.daviddesimone.opencloudsaves.yml:3]: Are you sure you need this? Elementary BaseApp is unmaintained, and it doesn't match the runtime/SDK you're using.

**Comment by barthalion** [comment]
bot, build io.github.daviddesimone.opencloudsaves

**Comment by DavidDeSimone** [comment]
I apologize for the build failure - the tag of my initial submission was not updated from v0.10.0 to v0.17.0

I have also updated the yml submission based on PR feedback.

File: io.github.daviddesimone.opencloudsaves.yml (hunk 2 above)
Comment (DavidDeSimone) [reply io.github.daviddesimone.opencloudsaves.yml]: Good catch - thank you. This is corrected.

File: io.github.daviddesimone.opencloudsaves.yml (hunk 1 above)
Comment (DavidDeSimone) [reply io.github.daviddesimone.opencloudsaves.yml]: Corrected.

**Comment by DavidDeSimone** [comment]
bot, build io.github.daviddesimone.opencloudsaves

**Comment by DavidDeSimone** [comment]
bot, build io.github.daviddesimone.opencloudsaves

File: io.github.daviddesimone.opencloudsaves.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
+app-id: io.github.daviddesimone.opencloudsaves
+runtime: org.gnome.Platform
+runtime-version: '42'
```
Comment (hfiguiere) [inline io.github.daviddesimone.opencloudsaves.yml:3]: this runtime will be EOL in may

43 is the current one.

File: io.github.daviddesimone.opencloudsaves.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+sdk: org.gnome.Sdk
+command: bootstrap-opencloudsave.sh
+cleanup:
+  - /app/lib/sdk/golang
+finish-args:
```
Comment (hfiguiere) [inline io.github.daviddesimone.opencloudsaves.yml:8]:
```suggestion
finish-args:
```

File: io.github.daviddesimone.opencloudsaves.yml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  - '--filesystem=host'
+  - '--socket=wayland'
+  - '--device=dri'
+  - '--share=network'
+build-options:
+  env:
+    - GOBIN=/app/bin
+    - GOROOT=/app/lib/sdk/golang
+modules:
+  - name: golang
```
Comment (hfiguiere) [inline io.github.daviddesimone.opencloudsaves.yml:20]: why this? There is a golang SDK

File: io.github.daviddesimone.opencloudsaves.yml (hunk 3 above)
Comment (DavidDeSimone) [reply io.github.daviddesimone.opencloudsaves.yml]: Thank you for the review - I have updated to 43.

File: io.github.daviddesimone.opencloudsaves.yml (hunk 4 above)
Comment (DavidDeSimone) [reply io.github.daviddesimone.opencloudsaves.yml]: I may be mistaken, but I think that this is no longer needed now that I am properly using the golang sdk extension.

File: io.github.daviddesimone.opencloudsaves.yml (hunk 5 above)
Comment (DavidDeSimone) [reply io.github.daviddesimone.opencloudsaves.yml]: I updated this to use the golang SDK extension. Earlier in my project I had a reason for doing it this way, but that reason is no longer relevant, and using the golang SDK extension works fine.

**Comment by DavidDeSimone** [comment]
bot, build io.github.daviddesimone.opencloudsaves

**Comment by barthalion** [comment]
bot, build io.github.daviddesimone.opencloudsaves

**Pull request closed without merging**